crates/bevy_core_widgets/src/core_radio.rs

```diff
@@ -0,0 +1,206 @@
+) {
+    if let Ok(CoreRadioGroup { on_change }) = q_group.get(ev.target()) {
+        let event = &ev.event().input;
+        if event.state == ButtonState::Pressed
+            && !event.repeat
+            && matches!(
+                event.key_code,
+                KeyCode::ArrowUp
+                    | KeyCode::ArrowDown
+                    | KeyCode::ArrowLeft
+                    | KeyCode::ArrowRight
+                    | KeyCode::Home
+                    | KeyCode::End
+            )
+        {
+            let key_code = event.key_code;
+            ev.propagate(false);
+
+            // Find all radio children that are not disabled
+            let radio_children = q_children
+                .iter_descendants(ev.target())
+                .filter_map(|child_id| match q_radio.get(child_id) {
+                    Ok((checked, false)) => Some((child_id, checked)),
+                    Ok((_, true)) | Err(_) => None,
+                })
+                .collect::<Vec<_>>();
+            if radio_children.is_empty() {
+                return; // No enabled radio buttons in the group
+            }
+            let current_index = radio_children
+                .iter()
+                .position(|(_, checked)| *checked)
+                .unwrap_or(usize::MAX); // Default to invalid index if none are checked
+
+            let next_index = match key_code {
+                KeyCode::ArrowUp | KeyCode::ArrowLeft => {
+                    // Navigate to the previous radio button in the group
+                    if current_index == 0 {
+                        // If we're at the first one, wrap around to the last
+                        radio_children.len() - 1
+                    } else {
+                        // Move to the previous one
+                        current_index - 1
+                    }
+                }
+                KeyCode::ArrowDown | KeyCode::ArrowRight => {
+                    // Navigate to the next radio button in the group
+                    if current_index >= radio_children.len() - 1 {
+                        // If we're at the last one, wrap around to the first
+                        0
+                    } else {
+                        // Move to the next one
+                        current_index + 1
+                    }
+                }
+                KeyCode::Home => {
+                    // Navigate to the first radio button in the group
+                    0
+                }
+                KeyCode::End => {
+                    // Navigate to the last radio button in the group
+                    radio_children.len() - 1
+                }
+                _ => {
+                    return;
+                }
+            };
+
+            if current_index == next_index {
+                // If the next index is the same as the current, do nothing
+                return;
+            }
+
+            let (next_id, _) = radio_children[next_index];
```

Comment:
Doesn't this panic if `current_index` was `usize::MAX` and `ArrowUp`/`ArrowLeft` was pressed?